test/forms/inputs/cloudinary-file-input/cloudinary-uploader.test.js

```diff
@@ -350,21 +515,35 @@ describe('cloudinaryUploader', () => {
 
     expect.assertions(1)
 
-    return expect(upload(fileData, file)).rejects.toThrow()
+    await act(async () => {
+      await expect(upload(fileData, file)).rejects.toThrow()
+    })
   })
 
-  test('updates the `uploadStatus` prop if request fails', () => {
-    const Wrapped = jest.fn(() => <h1>Hi</h1>)
+  test('updates the `uploadStatus` prop if request fails', async () => {
+    const Wrapped = (props) => (
+      <AsyncWrapper
+        fileData={fileData}
+        file={file}
+        {...props}
+        upload={(...args) => {
+          props.upload(...args).catch(() => {
+            // ignore thrown error
+          })
+        }}
+      />
+    )
     const Wrapper = cloudinaryUploader({ ...props, endpoint: '/failure' })(
       Wrapped
     )
+    const user = userEvent.setup()
     render(<Wrapper />)
-    const { upload } = Wrapped.mock.calls[0][0]
+    await act(async () => {
+      await user.click(screen.getByText('Upload'))
+    })
```

Comment:
I don't know why waiting for `upload-success` works without showing an act console error, but simply waiting for `upload-failure` does not work.

Let me know if you have any insights here!

Reply:
It probably doesn't help much, but I don't see any console errors removing the `act` wrapper here. Test passes without warnings.

Reply:
Honestly `act` warnings are my kryptonite. I can't believe this works now 😭. Thank you 🙏